lib/Parse/Lexer.cpp

```diff
@@ -1887,10 +1887,22 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
       } else if (!isMultilineString) {
         diagnose(CurPtr, diag::lex_unterminated_string);
         diagnose(CurPtr, diag::string_interpolation_unclosed);
+        diagnose(--TmpPtr, diag::opening_paren);
+
+        return formToken(tok::unknown, TokStart);
       } else if (*CurPtr == '\r' || *CurPtr == '\n') {
         diagnose(CurPtr, diag::lex_unterminated_string);
         diagnose(CurPtr, diag::string_interpolation_unclosed);
+        diagnose(--TmpPtr, diag::opening_paren);
+
+        wasErroneous = true;
+        continue;
         return formToken(tok::unknown, TokStart);
+      } else {
+        // as a fallback, just emit an unterminated string error
+        diagnose(CurPtr, diag::lex_unterminated_string);
+        wasErroneous = true;
+        continue;
```

Comment:
The fallback branch used to return `return formToken(tok::unknown, TokStart)`, let us preserve that. Also, we capitalize and punctuate `//` comments by convention.